MapboxDirections/MBLane.swift

```diff
@@ -1,125 +1,84 @@
-            type = .Left
-        case "right":
-            type = .Right
-        case "sharp left":
-            type = .SharpLeft
-        case "sharp right":
-            type = .SharpRight
-        case "slight left":
-            type = .SlightLeft
-        case "slight right":
-            type = .SlightRight
-        case "straight":
-            type = .StraightAhead
-        case "uturn":
-            type = .Uturn
-        case "none":
-            type = .None
-        default:
-            return nil
+    public init?(descriptions: [String]) {
+        var scope: LaneIndication = []
+        for description in descriptions {
+            switch description {
+            case "left":
+                scope.insert(.Left)
+            case "right":
+                scope.insert(.Right)
+            case "sharp left":
+                scope.insert(.SharpLeft)
+            case "sharp right":
+                scope.insert(.SharpRight)
+            case "slight left":
+                scope.insert(.SlightLeft)
+            case "slight right":
+                scope.insert(.SlightRight)
+            case "straight":
+                scope.insert(.StraightAhead)
+            case "uturn":
+                scope.insert(.Uturn)
+            case "none":
+                scope.insert(.None)
+            default:
+                return nil
+            }
         }
-        self.init(rawValue: type.rawValue)
+        self.init(rawValue: scope.rawValue)
     }
     
     public var description: String {
-        switch self {
-        case .Left:
-            return "left"
-        case .Right:
-            return "right"
-        case .SharpLeft:
-            return "sharp left"
-        case .SharpRight:
-            return "sharp right"
-        case .SlightLeft:
-            return "slight left"
-        case .SlightRight:
-            return "slight right"
-        case .StraightAhead:
-            return "straight"
-        case .Uturn:
-            return "uturn"
-        case .None:
-            return "None"
+        var descriptions: [String] = []
+        if contains(LaneIndication.Left) {
+            descriptions.append("Left")
+        }
+        if contains(LaneIndication.Right) {
+            descriptions.append("right")
         }
+        if contains(LaneIndication.SharpLeft) {
+            descriptions.append("sharp left")
+        }
+        if contains(LaneIndication.SharpRight) {
+            descriptions.append("sharp right")
+        }
+        if contains(LaneIndication.SlightLeft) {
+            descriptions.append("slight left")
+        }
+        if contains(LaneIndication.SlightRight) {
+            descriptions.append("slight right")
+        }
+        if contains(LaneIndication.StraightAhead) {
+            descriptions.append("straight")
+        }
+        if contains(LaneIndication.Uturn) {
+            descriptions.append("uturn")
+        }
+        if contains(LaneIndication.None) {
```

Comment:
These are the only values we return:

https://github.com/Project-OSRM/osrm-backend/blob/master/docs/http.md#properties-5

`||right` and `none|none|right` are the same for OSRM and will be converted to the fixed set of return types in the spec above.